### ISSUE TITLE
Configuration changes in 2.0

### ISSUE THREAD

**Issue by tdykstra** [aspnet/Docs#3387]
Affects fundamentals/configuration.md and others.
Document Options 2.0 changes #3469

**Comment by Pomoinytskyi** [comment]
Hi all.
I created project from web template (Asp.Net core 2.0) and found that signature of Startup constructor was changed. Now it receives IConfiguration instead of IHostingEnvironment. Can you reference me to the document with best practice how to  add UserSecrets to the configuration.
Thanks.

**Comment by guardrex** [comment]
@rachelappel There's a question that's come up while I was working these config samps. Not working this topic; just updating the samps to 2.0 on PRs: #4197, #4198, #4199, #4200, #4201, #4205, #4206, #4207 (all to address getting rid of `dnxcore50` on #1950) ... and three upcoming PRs on Tue 9/5 (no `dnxcore50`, but I may as well ... *I'm on a roll!* :smile:). One sample, `IOptionsSnapshot2` is outside of the *src* folder. Not sure why that was placed there, but it's also unclear if you want to move all of the samps out of *src* and into the *sample* folder. I'll (quickly) get the samps updated, then get out of your way here.

![capture10](https://user-images.githubusercontent.com/1622880/30041309-acd5efb4-91ae-11e7-88a1-25c601d1a353.PNG)

[Dots indicate orphaned samps that are slated for obliteration.]

**Comment by scottaddie** [comment]
@guardrex @rachelappel I recommend moving all the sample code under "sample" and deleting the "src" folder. The "src" folder was used in VS 2015 project templates. Since it isn't used in the VS 2017 project templates, it's probably best to move away from it.

**Comment by guardrex** [comment]
Very good. There was one other one in my batch for #1950 that had has an (unnecessary) *src*+*test* folder layout. IIRC the test projects are well-named, so they can sit in the *sample* folder next to the project folder. I've made a note here to circle back around to it.

This batch is a little different in that the *src* folder holds completely separate projects ... not test vs. app ... so yeah ... it makes very little sense here at all, even under VS2015.

I'll take care of the other one, but I'll leave these Configuration projects in the *src* to be worked on this PR.

There are three here that need upgrade to 2.0, and I'll get them today right after I get back from the 💪 gym and a quick 🚴 bike ride.

**Comment by guardrex** [comment]
Cross-ref: https://github.com/aspnet/Configuration/issues/728

... but it isn't 100% clear if this the tracking topic for *ALL* configuration topic updates for 2.0 (e.g., `ConfigureAppConfiguraton`).

**Comment by scottaddie** [comment]
@guardrex I'll be adding a small section to the existing migration doc to outline ONLY major breaking changes. (see issue https://github.com/aspnet/Docs/issues/4277). This issue should remain open, and the resulting doc should serve as a much more detailed reference.

**Comment by guardrex** [comment]
@rachelappel I'm looking at https://github.com/aspnet/Docs/issues/1382 and wondering if you think I have time to sneak a quick-ish (24-48 hours) PR in there to cover that before you go to work on this issue?

**Comment by rachelappel** [comment]
@guardrex  Yep. I am working on another issue at the moment and have another I can bump up in the queue in the meantime so no problem there.

**Comment by guardrex** [comment]
@scottaddie Ok with you? ... kills two :bird: :bird: ... that issue and wraps up https://github.com/aspnet/Docs/issues/2409. Be nice if [@]Rick-Anderson comes back to fewer issues than he left (if that's possible lol).

**Comment by scottaddie** [comment]
@guardrex Sounds good to me!

**Comment by guardrex** [comment]
@rachelappel Update on https://github.com/aspnet/Docs/pull/4307 The work is done ... just waiting on review. I'll jump on feedback asap to get out of your way here.

**Comment by guardrex** [comment]
@Rick-Anderson Since options are part of the config topic and probably will be addressed at the same time, should #3469 be closed and tracking that work be done over here?
